pkgs/tools/misc/abduco/default.nix

```diff
@@ -1,14 +1,20 @@
-{ lib, stdenv, fetchFromGitHub, writeText, conf ? null }:
+{ lib, stdenv, fetchpatch, fetchzip, writeText, conf ? null }:
 
-stdenv.mkDerivation rec {
-  pname = "abduco";
+let
+  name = "abduco";
+  owner = "martanne";
+  rev = "8c32909a159aaa9484c82b71f05b7a73321eb491";
+in
+stdenv.mkDerivation {
+  pname = name;
   version = "2020-04-30";
 
-  src = fetchFromGitHub {
-    owner = "martanne";
-    repo = "abduco";
-    rev = "8c32909a159aaa9484c82b71f05b7a73321eb491";
-    sha256 = "0a3p8xljhpk7zh203s75248blfir15smgw5jmszwbmdpy4mqzd53";
+  src = fetchzip {
+    urls = [
+      "https://github.com/${owner}/${name}/archive/${rev}.tar.gz"
+      "https://git.sr.ht/~${owner}/${name}/archive/${rev}.tar.gz"
```

Comment:
Please do not substitute all parts of the URL and just write them literal.

Also fetching from GitHub and SourceHut should use the corrosponding fetcher which also does not need to be substituted.

Reply:
Well, no. Those are not straight up mirrors and the exact hash can depend on the git version used.